chcemvediet/apps/inforequests/tests/test_views/test_mine.py

```diff
@@ -0,0 +1,138 @@
+        drafts2 = [self._create_inforequest_draft(applicant=self.user2) for i in range(3)]
+        pending_inforequests2 = [self._create_inforequest(applicant=self.user2) for i in range(5)]
+        successful_inforequests2 = [self._create_successful_inforequest(applicant=self.user2) for i in range(3)]
+        unsuccessful_inforequests2 = [self._create_inforequest(applicant=self.user2, closed=True) for i in range(3)]
+
+        self._login_user(self.user1)
+        response = self.client.get(reverse(u'inforequests:mine'))
+        self.assertEqual(response.status_code, 200)
+        self.assertItemsEqual(response.context[u'pending_inforequests'], [])
+        self.assertItemsEqual(response.context[u'drafts'], [])
+        self.assertItemsEqual(response.context[u'successful_inforequests'], [])
+        self.assertItemsEqual(response.context[u'unsuccessful_inforequests'], [])
+
+    def test_related_models_are_prefetched_before_render(self):
+        drafts1 = [self._create_inforequest_draft(applicant=self.user1) for i in range(5)]
+        pending_inforequests1 = [self._create_inforequest(applicant=self.user1) for i in range(4)]
+        successful_inforequests1 = [self._create_successful_inforequest(applicant=self.user1) for i in range(3)]
+        unsuccessful_inforequests1 = [self._create_inforequest(applicant=self.user1, closed=True) for i in range(3)]
+
+        # Force view querysets to evaluate before calling render
+        def pre_mock_render(request, template, context):
+            list(context[u'pending_inforequests'])
+            list(context[u'drafts'])
+            list(context[u'successful_inforequests'])
+            list(context[u'unsuccessful_inforequests'])
+
+        self._login_user(self.user1)
+
+        # main/base/single_column.html
+        patterns = [
+            u'FROM "accounts_profile" WHERE "accounts_profile"."user_id" = %s',
+            squeeze(u"""
+                FROM "mail_message"
+                INNER JOIN "inforequests_inforequestemail" ON \( "mail_message"."id" = "inforequests_inforequestemail"."email_id" \)
+                INNER JOIN "inforequests_inforequest" ON \( "inforequests_inforequestemail"."inforequest_id" = "inforequests_inforequest"."id" \)
+                WHERE \("inforequests_inforequestemail"."type" = %s
+                    AND "inforequests_inforequest"."applicant_id" = %s
+                    AND "inforequests_inforequest"."closed" = %s\)
+                """),
+            u'FROM "invitations_invitationsupply" WHERE "invitations_invitationsupply"."user_id" = %s',
+        ]
+
+        # pending_inforequests
+        patterns.extend([
+            squeeze(u"""
+                FROM "inforequests_action"
+                INNER JOIN "inforequests_branch" ON \( "inforequests_action"."branch_id" = "inforequests_branch"."id" \)
+                WHERE "inforequests_branch"."inforequest_id" = %s
+                """) for _ in range(4)
```

Comment:
@martinmacko47 Tych queries je strasne vela, preto som ich nedefinoval vsetky naraz v jednom kroku, ale pouzil som `extend` a pridal koment odkial queries pochadzaju.
Viem ze v testoch by sa malo radsej vypisat vsetko  explicitne ale, bolo by to vela pisania, preto som si pomohol s `for _ in range(4)` alebo `* 3`.

Reply:
A pride mi tiez zvlastne ze predtym v tomto teste sa neasertoval ani jeden query `with self.assertQueriesDuringRender([], pre_mock_render=pre_mock_render):` a teraz je tych queries vyse 20, pricom `inforequest_mine` view a prisluchajuci template sa nezmenil uplne vo vsetkom, niketore veci ostali rovnake.